### PR TITLE
[DO NOT MERGE] added nix flake

### PR DESCRIPTION
**Pull request by isubasinghe**
Waiting on adding support for macOS in the nixified compiler here:
https://github.com/isubasinghe/arm-gcc-binutils

### CODE REVIEW

**Comment by wucke13** [comment]
@isubasinghe Please take a look on https://github.com/DLR-FT/seL4-nix-utils. I think I did (almost?) all the work you intend to do here.

**Comment by isubasinghe** [comment]
Thanks for letting me know @wucke13, in that case I will create some PRs there for anything missing there.

**Comment by wucke13** [comment]
@isubasinghe Great, I'm looking forward to your contributions :smile:. I might suggest you close this PR then, if I understand correctly it will not be pursued further in the near future?

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/24.05";
+    utils.url = "github:numtide/flake-utils";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+  };
+
+  outputs = { self, nixpkgs, rust-overlay, ... }@inputs: inputs.utils.lib.eachSystem [
+    # Add the system/architecture you would like to support here. Note that not
+    # all packages in the official nixpkgs support all platforms.
+    "x86_64-linux" "i686-linux" "aarch64-linux" "x86_64-darwin"
```
Comment (Ivan-Velickovic) [inline flake.nix:13]:
```suggestion
    "x86_64-linux" "aarch64-linux" "x86_64-darwin"
```

**Comment by Ivan-Velickovic** [comment]
> @isubasinghe Great, I'm looking forward to your contributions 😄. I might suggest you close this PR then, if I understand correctly it will not be pursued further in the near future?

I could not find how to get started with https://github.com/DLR-FT/seL4-nix-utils and we need something simple/quick for the Microkit CI so I think I might merge this since it's mostly ready. It's only ~100 lines so I think it's fine to have here on it's own.

**Comment by wucke13** [comment]
As you like. If I can be of any help, i.e. explaining the stuff from our repo, please let me know.

**Comment by Ivan-Velickovic** [comment]
Thanks @isubasinghe for starting this, I've made https://github.com/seL4/microkit/pull/233 to finish it off.

**Pull request closed without merging**